### PR TITLE
Adds --all flag to test to test all functions in an app. Adds walkFun…

### PR DESCRIPTION
**Pull request by treeder**
…c method and colors too.

Colors and final output look like:

screen shot 2017-12-11 at 3 49 55 pm

### CODE REVIEW

File: run.go [hunk 1]
```diff
@@ -67,7 +67,8 @@ func runflags() []cli.Flag {
 	}
 }
 
-func preRun(c *cli.Context) (*funcfile, []string, error) {
+// preRun parses func.yaml, checks expected env vars and builds the function image.
+func preRun(c *cli.Context) (string, *funcfile, []string, error) {
```
Comment (rdallman) [inline run.go]: nit: probably time to return an object here, without named params this is gobbledy

File: walker.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// WalkFuncsFunc good name huh?
```
Comment (rdallman) [inline walker.go]: say this 10 times really fast

File: walker.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+// walkFuncs is similar to filepath.Walk except only returns func.yaml's (so on per function)
+func walkFuncs(root string, walkFn walkFuncsFunc) error {
+	return filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			// logging this so we can figure out any common issues
+			fmt.Println("error walking filesystem:", err)
+			return err
+		}
+		// was `path != wd` necessary?
```
Comment (rdallman) [inline walker.go]: don't think so, since wd should always be a dir

**Comment by treeder** [comment]
Looks like I'm getting this: https://github.com/golang/go/issues/22688

**Comment by rdallman** [comment]
glide install every build ftw ?

**Comment by treeder** [comment]
dep for the win now. ;)

**Comment by rdallman** [comment]
talladega-knife-leg

**Comment by treeder** [comment]
Annnnd, another fun one.  Guess the docker update that required the new golang/sys now needs gcc or on the machine or something? 

```
Status: Downloaded newer image for golang:alpine
# github.com/fnproject/cli/vendor/github.com/docker/docker/pkg/term
exec: "gcc": executable file not found in $PATH
```

I even get this locally now too when I run: `docker run --rm -v ${PWD}:/go/src/github.com/fnproject/cli -w /go/src/github.com/fnproject/cli golang:alpine go build -o fn_alpine`

**Pull request merged**